nanoAOD/plugins/HadTruthProducer.cc

```diff
@@ -82,7 +88,49 @@ HadTruthProducer::produce( edm::Event& iEvent, const edm::EventSetup& iSetup)
   hadTruthTable->addColumn<int>("isHadFromTsb",isHadFromTsb,"Hadron from t->s/b",nanoaod::FlatTable::IntColumn);
   hadTruthTable->addColumn<uint8_t>("isHadFromTop",isHadFromTop,"Hadron from Top",nanoaod::FlatTable::UInt8Column);  
   iEvent.put(move(hadTruthTable),"hadTruth");
-  
+
+  std::vector<int> idx;
+  std::vector<int> pdgId;
+  std::vector<uint8_t> isMatched;
+  std::vector<int> isMatHadFromTsb;
+  std::vector<uint8_t> isMatHadFromTop;
+
+  for (unsigned int i = 0; i < genParticles.product()->size(); ++i) {
```

Comment:
Shouldn't we be adding this information to the genHadron table, rather than starting a new one?